artifactory/commands/transferfiles/state/utils.go

```diff
@@ -121,3 +122,27 @@ func GetOldTransferDirectoryStructureError() error {
 	}
 	return errorutils.CheckErrorf(oldTransferDirectoryStructureErrorFormat, transferDir)
 }
+
+// Atomically add to an int64 variable.
+// addr     - Pointer to int64 variable
+// delta    - The change to do
+// increase - True to increment, false to decrement
+func AtomicallyAddInt64(addr *int64, delta int64, increase bool) {
```

Comment:
I believe this function can be made provate.
The same goes for the `AtomicallyAddUint64` function.